build-system/tasks/dist.js

```diff
@@ -205,23 +201,20 @@ function buildLoginDone(version) {
  */
 async function buildWebPushPublisherFiles() {
   const distDir = 'dist/v0';
-  const promises = [];
-  WEB_PUSH_PUBLISHER_VERSIONS.forEach((version) => {
-    WEB_PUSH_PUBLISHER_FILES.forEach((fileName) => {
-      const tempBuildDir = `build/all/amp-web-push-${version}/`;
-      const builtName = fileName + '.js';
-      const minifiedName = maybeToEsmName(fileName + '.js');
-      const p = compileJs('./' + tempBuildDir, builtName, './' + distDir, {
+  for (const version of WEB_PUSH_PUBLISHER_VERSIONS) {
+    for (const fileName of WEB_PUSH_PUBLISHER_FILES) {
+      const tempBuildDir = `build/all/amp-web-push-${version}`;
+      const builtName = `${fileName}.js`;
+      const minifiedName = maybeToEsmName(builtName);
+      await compileJs(`./${tempBuildDir}`, builtName, `./${distDir}`, {
```

Comment:
Why did you change the pattern here? Isn't it slower to waterfall the compilations instead of parallelizing via Promise.all?

Reply:
I did think of this, and good of you to flag it! I switched the pattern primarily for better readability and predictability of order, and because performance is actually better here (because we're not using vinyl fs). Predictable order helps when we enable debugging for compilation (`gulp dist --debug`).

Here's the reference I used to research this: https://medium.com/dailyjs/the-pitfalls-of-async-await-in-array-loops-cf9cf713bfeb

Reply:
Update: I've parallelized all file copies that no longer need to be serial. There are one or two instances like compiling two related parts of the same module that still happen one after the other. Nice feedback!